src/EditorFeatures/CSharpTest/Completion/CompletionProviders/IndexerCompletionProviderTests.cs

```diff
@@ -0,0 +1,502 @@
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task IndexerOfBaseTypeIsNotSuggestedIfNotAccessible()
+        {
+            await VerifyNoItemsExistAsync(@"
+public class Base
+{
+    protected int this[int i] => i;
+}
+public class Derived : Base
+{
+}
+
+public class Program
+{
+    public static void Main()
+    {
+        var d = new Derived();
+        d.$$
+    }
+}
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task IndexerIsSuggestedOnString()
+        {
+            await VerifyItemExistsAsync(@"
+public class Program
+{
+    public static void Main()
+    {
+        var s = ""Test"";
+        s.$$
+    }
+}
+", "this", displayTextSuffix: "[]");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task TestEditorBrowsableOnIndexerIsRespected_EditorBrowsableStateNever()
+        {
+            var markup = @"
+namespace N
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            var c = new C();
+            c.$$
+        }
+    }
+}
+";
+            var referencedCode = @"
+using System.ComponentModel;
+
+namespace N
+{
+    public class C
+    {
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int this[int i] => i;
+    }
+}
+";
+
+            await VerifyItemInEditorBrowsableContextsAsync(
+                markup: markup,
+                referencedCode: referencedCode,
+                item: "this",
+                expectedSymbolsSameSolution: 1,
+                expectedSymbolsMetadataReference: 0,
+                sourceLanguage: LanguageNames.CSharp,
+                referencedLanguage: LanguageNames.CSharp);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task TestEditorBrowsableOnIndexerIsRespected_EditorBrowsableStateAdvanced()
+        {
+            var markup = @"
+namespace N
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            var c = new C();
+            c.$$
+        }
+    }
+}
+";
+            var referencedCode = @"
+using System.ComponentModel;
+
+namespace N
+{
+    public class C
+    {
+        [EditorBrowsable(EditorBrowsableState.Advanced)]
+        public int this[int i] => i;
+    }
+}
+";
+
+            await VerifyItemInEditorBrowsableContextsAsync(
+                markup: markup,
+                referencedCode: referencedCode,
+                item: "this",
+                expectedSymbolsSameSolution: 1,
+                expectedSymbolsMetadataReference: 0,
```

Comment:
This test is based on https://github.com/dotnet/roslyn/blob/ad668af283ac619c0076766ba0db0df83aaf06f7/src/EditorFeatures/CSharpTest/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProviderTests.cs#L897

The test is failing, because `this[]` is suggested despite `hideAdvancedMembers: true`. I haven't looked into this. @CyrusNajmabadi Do you have a look?

Reply:
sure. i'll look.  it's not that compelling an area for me though :)

Reply:
ok.  i got this test working.  thanks!